app/models/question.rb

```diff
@@ -3,4 +3,49 @@ class Question < ApplicationRecord
   belongs_to :creator, class_name: User.name, foreign_key: :user_id
   has_many :answers, dependent: :destroy
   has_many :test_questions, dependent: :destroy
+  accepts_nested_attributes_for :answers,
+                                reject_if: :all_blank,
+                                allow_destroy: true
+
+  enum question_type: {single_choice: 0, multiple_choice: 1}
+
+  validates :content, presence: true
+  validates :question_type, presence: true, inclusion: {in: question_types.keys}
+  validates_associated :answers
+  # Toi thieu 2 lua chon
+  validates :answers, length: {minimum: Settings.digit.length_2,
+                               too_short: I18n.t("errors.minimum_answers")}
+  validate :must_have_one_correct_answer, on: :create
+  validate :validate_single_type, on: :create, if: :is_single_choice?
```

Comment:
```suggestion
  validate :validate_single_type, on: :create, if: self.single_choice?
```
ri dc ko em, tận dụng enum thì ko cần tạo thêm 2 method `is_single_choice?` và `:is_multiple_choice?` nữa